chatarena/environments/conversation.py

```diff
@@ -121,31 +144,162 @@ def step(self, player_name: str, action: str) -> TimeStep:
+
+        if isinstance(moderator, AgentConfig):
+            moderator_config = moderator
+            moderator = Moderator.from_config(moderator_config)
+        elif not isinstance(moderator, Moderator):
+            raise ValueError(
+                "moderator must be either an AgentConfig or a Moderator instance."
+            )
+
+        self.moderator = moderator
+
+        self._question = _lower_first_letter(question if question else DEFAULT_QUESTION)
+        self._answer_1 = _lower_first_letter(answer_1 if answer_1 else DEFAULT_ANSWER_1)
+        self._answer_2 = _lower_first_letter(answer_2 if answer_2 else DEFAULT_ANSWER_2)
+        self._max_num_rounds = max_num_rounds if max_num_rounds else DEFAULT_NUM_ROUNDS
+        assert len(player_names) == 2
+
+        self._initial_messages = [
+            Message(
+                agent_name=self.moderator.name,
+                content=f"The question of today's debate is {self._question}",
+                turn=0,
+            ),
+            Message(
+                agent_name=self.player_names[0],
+                content=f"I claim the answer is {self._answer_1}",
+                turn=0,
+            ),
+            Message(
+                agent_name=self.player_names[1],
+                content=f"I claim the answer is {self._answer_2}",
+                turn=0,
+            ),
+        ]
+
+    def reset(self):
+        init_timestep = super().reset()
+        for message in self._initial_messages:
+            self.message_pool.append_message(message)
+        self._current_turn = 1
+        return init_timestep
+
+    def to_config(self) -> EnvironmentConfig:
+        # This environment contains some speical config arguments that needs to be handle specially
+        return EnvironmentConfig(
+            env_type=self.type_name,
+            player_names=self.player_names,
+            moderator=self.moderator.to_config(),
+            parallel = self.parallel,
+            question=self._question,
+            answer_1=self._answer_1,
+            answer_2=self._answer_2,
+            max_num_rounds=self._max_num_rounds,
+        )
+
+    def step(self, player_name: str, action: str) -> TimeStep:
+        """
+        step function that is called by the arena
+        Args:
+            player_name: the name of the player that takes the action
+            action: the action that the agents wants to take
+        """
+        message = Message(
+            agent_name=player_name, content=action, turn=self._current_turn, visible_to=player_name
+        )
+        self.message_pool.append_message(message)
+
+        # Round-robin order for the next player
+        self._next_player_idx = (self._next_player_idx + 1) % self.num_players
+        terminal = self.is_terminal()
+
+        # Update for end of round
+        if not self.parallel or self._next_player_idx == 0:
+            for message in self.message_pool.get_all_messages()[-self.num_players:]:
```

Comment:
Players should only see responses from previous rnds e.g. Player 2 can't see Player 1's response from rnd 2 in their rnd 2 response

This is a bit of a bodge and could probably be handled more neatly with an custom MessagePool data structure